### ISSUE TITLE
Automating the website build process

### ISSUE THREAD

**Issue by siavashserver** [glfw/website#77]
Since we aren't using any fancy Jekyll plugins, we can let GitHub to directly build and host the website by moving the sources to `glfw.github.io` repository.

**Comment by elmindreda** [comment]
That would be so good to have working.

Should the Doxygen generated directories live in this repo as well?

**Comment by siavashserver** [comment]
Yes, so they can be copied over during build process as well.

**Comment by elmindreda** [comment]
Does e53f035fb1ece2766fb27c35a9adba87a6268e29 seem reasonable?

**Comment by siavashserver** [comment]
> Does [e53f035](https://github.com/glfw/website/commit/e53f035fb1ece2766fb27c35a9adba87a6268e29) seem reasonable?

Yup.

I have also made a few changes here a3f2ff925cc9e471c3ca45f5639aa258d93af55b to fix issues with stylesheet paths; that's probably due to *GitHub Pages* stack being different (Jekyll 3.9) .

I'm not really familiar with how custom domain binding works, you might need to somehow copy/keep `CNAME` in sources as well.

**Comment by elmindreda** [comment]
Yes, the `CNAME` file needs to move here and possibly we should also rename this repo to `glfw.github.io`? I'm not sure what the functional differences are between a project site and an organization site beyond naming, but this is technically the latter.

**Comment by siavashserver** [comment]
I guess that's all that needs to be done and we should be good to go ^^"

**Comment by elmindreda** [comment]
It worked perfectly! Awesome work!

**Comment by elmindreda** [comment]
~Hmm, nearly. `css/arrow.png` isn't getting copied for some reason.~ Never mind, it's back now.

**Comment by siavashserver** [comment]
I love these refreshing *6:00 AM* mini heart attacks, glad it's up and running on first try :D

I'm not sure how much useful it might be, but it seems like that documentation regeneration can be somehow automated through these GitHub Actions as well:
- [Doxygen Action](https://github.com/marketplace/actions/doxygen-action)
- [GitHub Pages action](https://github.com/marketplace/actions/github-pages-action)
- [Deploying to external repositories](https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-deploy-to-external-repository-external_repository)
